kernel/src/log_compaction/mod.rs

```diff
@@ -0,0 +1,102 @@
+//! # Log Compaction
+//!
+//! This module provides an API for writing log compaction files that aggregate
+//! multiple commit JSON files into single compacted files. This improves performance
+//! by reducing the number of individual log files that need to be processed during
+//! log replay operations.
+//!
+//! ## Overview
+//!
+//! Log compaction creates files with the naming pattern `{start_version}.{end_version}.compacted.json`
+//! that contain the reconciled actions from all commit files in the specified version range.
+//! Only commit/compaction files that intersect with [start_version, end_version] are processed.
+//! This is similar to checkpoints but operates on a subset of versions rather than the entire table.
+//!
+//! ## Usage
+//!
+//! The log compaction API follows a similar pattern to the checkpoint API:
+//!
+//! 1. Create a [`LogCompactionWriter`] using [`crate::Snapshot::get_log_compaction_writer`] to compact the log
+//!    from a given start_version to end_version (inclusive)
+//! 2. Get the compaction path from [`LogCompactionWriter::compaction_path`]
+//! 3. Get the compaction data from [`LogCompactionWriter::compaction_data`]
+//! 4. Write the data to the path in cloud storage (engine-specific)
+//!
+//! ## Example
+//!
+//! ```no_run
+//! # use std::sync::Arc;
+//! # use delta_kernel::log_compaction::{LogCompactionDataIterator, LogCompactionWriter};
+//! # use delta_kernel::{Engine, Snapshot, DeltaResult, Error, FileMeta};
+//! # use url::Url;
+//!
+//! // Engine-specific function to write compaction data
+//! fn write_compaction_file(path: Url, data: LogCompactionDataIterator) -> DeltaResult<FileMeta> {
+//!     // In a real implementation, this would write the data to cloud storage
+//!     todo!("Write data batches to storage at path: {}", path)
+//! }
+//!
+//! # fn example(engine: &dyn Engine) -> DeltaResult<()> {
+//! // Create a snapshot for the table
+//! let table_root = url::Url::parse("file:///path/to/table")?;
+//! let snapshot = Arc::new(Snapshot::builder(table_root).build(engine)?);
+//!
+//! // Create a log compaction writer for versions 10-20
+//! let mut writer = snapshot.get_log_compaction_writer(10, 20)?;
+//!
+//! // Get the compaction path and data
+//! let compaction_path = writer.compaction_path().clone();
```

Comment:
shouldn't need clone right?
```suggestion
//! let compaction_path = writer.compaction_path();
```